Feat/overscanning

satyamgaur opened the pull request:
I want to discuss one thing, that I've used an additional param in `getVisibleVertices` and `getVisibleEdges`, `overscan`, with the help of which we can toggle the overscan feature if we want. And to implement overscanning for us I've declared another variable on _line 378_ inside the `renderChildren` function.. But I guess it would be much better if we would declare that variable in the constructor itself. But wanted to discuss it first before implementing.

satyamgaur committed:
feat: adds static overscanning

satyamgaur committed:
Revert "feat: adds static overscanning"

This reverts commit eac7005be8453ebd9d11bac8e95bb68bee8d31dd.

Correction: changed single quotes to double quotes

satyamgaur committed:
Feat: Adds static overscanning

smitpatel1010 commented on .prettierrc:
```diff
@@ -1,6 +1,6 @@
-  "singleQuote": true,
-  "trailingComma": "es5"
-}
+  "printWidth": 80,
```
revert changes from this file

satyamgaur replied:
done

smitpatel1010 commented on src/Diagram.js:
```diff
@@ -368,8 +375,9 @@ class Diagram extends React.PureComponent {
   renderChildren(extremeX, extremeY, zoom) {
-    const verticesMap = this.getVisibleVertices(zoom);
-    const edges = this.getVisibleEdges(zoom);
+    const overscan = 1;
```
lets take a prop `overscanPixels` as a number. and fallback of that should be width and height of the container

satyamgaur replied:
done

satyamgaur committed:
reverts changes in .prettierrc and makes use of overscanPixels as a prop

milap1296 commented on src/Diagram.js:
```diff
@@ -280,9 +252,16 @@ class Diagram extends React.PureComponent {
       : DEFAULT_CONTAINER_RECT;
+    const viewport = getViewport(scroll.left, scroll.top, width, height, zoom);
+    const viewportWithOverscanning = {
+      xMin: viewport.xMin - (this.props.overscanPixels ? this.props.overscanPixels.width : width),
```
this.props.overscanPixels this can be stored in a variable instead of doing it multiple times

satyamgaur replied:
will take care of it in the next commit :)

satyamgaur committed:
cleans code and adds PropTypes for overscan

removes unnecessary this.props.overscanPixels

adds PropTypes for overscan and adds nullish coalescing operator

smitpatel1010 commented on src/Diagram.js:
```diff
@@ -280,9 +252,16 @@ class Diagram extends React.PureComponent {
+      xMin: viewport.xMin - (this.props.overscanPixels ? this.props.overscanPixels.width : width),
+      xMax: viewport.xMax + (this.props.overscanPixels ? this.props.overscanPixels.width : width),
+      yMin: viewport.yMin - (this.props.overscanPixels ? this.props.overscanPixels.height : height),
+      yMax: viewport.yMax + (this.props.overscanPixels ? this.props.overscanPixels.height : height),
```
add `@babel/plugin-proposal-nullish-coalescing-operator` in `.babelrc` `plugins`
and then use `??`

satyamgaur replied:
done

smitpatel1010 replied:
@satyamgaursprinklr need to add this in package.json as a devDependency too

satyamgaur committed:
adds nullish coalescing as devDependencies in pacakge.json

smitpatel1010 commented on package.json:
```diff
@@ -21,6 +21,7 @@
     "@babel/plugin-proposal-class-properties": "7.7.4",
     "@babel/plugin-proposal-optional-chaining": "^7.16.7",
     "@babel/plugin-transform-react-jsx-source": "7.7.4",
+    "@babel/plugin-proposal-nullish-coalescing-operator": "7.17.12",
```
Have you ran `yarn install` after adding in `package.json`?

satyamgaur replied:
Oh, I wasn't aware of that, are there any more changes, I'll push all of them together in a single commit.

satyamgaur replied:
done

satyamgaur committed:
adds null coalescing in yarn.lock and changes width and height in overscan prop to x and y
